File: fibonacci.rb
#! /usr/bin/env ruby

$stdout.sync = true

def fib(rot)
  
  old_value = 1
  new_value = 0

  while rot > 0 do
    old_value, new_value = new_value, new_value + old_value
    rot -= 1
  end

  print new_value
  
end

# If no arguments, run 20 fibs:
if ARGV.length == 0 then
  (0...10).each {|i| fib(i); print "\t"; fib(i + 10); print "\n"}

# Else, fib the first argument:
else
  fib(ARGV[0].to_i)
  puts
end

=begin
Name: Fibonacci Sequence
Language: Ruby
State: Done

Prints out numbers from the fibonacci sequence


Example: ./fibonacci.rb
Example2: ./fibonacci.rb 42
=end

File: language-testing/fibonacci/se/iix/Fibonacci.java
package se.iix;

/**
 * Fibonacci
 * Prints out numbers from the fibonacci sequence
 */
public class Fibonacci {
  public static void main(String[] args) {
    /* If there's an argument, fib it!
     * else, run 20 fibs */
    if (args.length > 0) {
      System.out.println(fib(Integer.parseInt(args[0])));
    } else {
      for (int i = 0; i < 10; i++) {
        System.out.println(fib(i) + "\t" + fib(i+10));
      }
    }
  }

  public static int fib(int n) {
    int tmp;
    int old_value = 1;
    int new_value = 0;
    for (int i = 0; i < n; ++i) {
      tmp = old_value;
      old_value = new_value;
      new_value = tmp + old_value;
    }
    return new_value;
  }
}
